### PR TITLE
Track user progress

### PR DESCRIPTION
**Pull request by aziz512**
This directive sends current user slide and the time user spend on different slides

### CODE REVIEW

**Comment by kirjs** [comment]
Fix travis build?

File: src/app/tracking/tracking.directive.spec.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { TrackingDirective } from './tracking.directive';
+
+describe('TrackingDirective', () => {
+  it('should create an instance', () => {
```
Comment (prestonvanloon) [inline src/app/tracking/tracking.directive.spec.ts]: I don't think this is a useful test. Can you test the compiled angular directive? We have an example here: https://github.com/AngularNYC/angular-presentation/blob/master/src/app/presentation/mode-routing/mode-routing.directive.spec.ts

File: src/app/tracking/tracking.directive.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    afAuth.auth.signInAnonymously();
+    afAuth.authState.subscribe(authData => {
+      this.auth = authData;
+    });
+    this.lastSlideChange = Date.now();
+  }
+
+  @HostListener('onSlideChange', ['$event']) onSlideChange(index) {
+    if (this.auth) {
+      const diffMinutes = Date.now() - this.lastSlideChange;
```
Comment (prestonvanloon) [inline src/app/tracking/tracking.directive.ts]: You're calling `Date.now()` three times in this method. Consider adding a const and using it three times.

File: src/app/tracking/tracking.directive.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+              private router: Router, private presentation: PresentationComponent) {
+    afAuth.auth.signInAnonymously();
+    afAuth.authState.subscribe(authData => {
+      this.auth = authData;
+    });
+    this.lastSlideChange = Date.now();
+  }
+
+  @HostListener('onSlideChange', ['$event']) onSlideChange(index) {
+    if (this.auth) {
```
Comment (kirjs) [inline src/app/tracking/tracking.directive.ts]: What's the case where we don't have auth?

**Comment by kirjs** [comment]
Looks like everything is fixed

**Pull request merged**